WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerBottomSheet.kt

```diff
@@ -0,0 +1,181 @@
+        super.onCreate(savedInstanceState)
+        (requireActivity().applicationContext as WordPress).component().inject(this)
+        viewModel = ViewModelProvider(requireActivity(), viewModelFactory).get(LocalePickerViewModel::class.java)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = LocalePickerBottomSheetBinding.inflate(inflater, container, false)
+        return binding?.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding?.apply {
+            setupContentViews()
+            setupObservers()
+        }
+    }
+
+    private fun LocalePickerBottomSheetBinding.setupContentViews() {
+        val layoutManager = LinearLayoutManager(context)
+        recyclerView.layoutManager = layoutManager
+        recyclerView.addItemDecoration(DividerItemDecoration(context, RecyclerView.VERTICAL))
+        recyclerView.adapter = localeAdapter
+        recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                if (newState == RecyclerView.SCROLL_STATE_DRAGGING) {
+                    viewModel.onListScrolled()
+                }
+            }
+        })
+
+        searchInputLayout.editText?.doOnTextChanged { _, _, _, _ ->
+            onSearchStatusUpdated()
+        }
+
+        searchInputLayout.editText?.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+            if (hasFocus) {
+                viewModel.onSearchFieldFocused()
+            }
+        }
+
+        searchInputLayout.editText?.setOnKeyListener { _, keyCode, event ->
+            if (event.action == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_ENTER) {
+                onSearchStatusUpdated()
+                true
+            } else {
+                false
+            }
+        }
+
+        searchInputLayout.editText?.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_GO) {
+                onSearchStatusUpdated()
+                true
+            } else {
+                false
+            }
+        }
+
+        searchInputLayout.setEndIconOnClickListener {
+            viewModel.onClearSearchFieldButtonClicked()
+        }
+
+        btnLocaleSuggestion.setOnClickListener {
+            viewModel.onCurrentLocaleSelected()
+        }
+
+        dialog?.setOnShowListener { dialogInterface ->
+            val sheetDialog = dialogInterface as? BottomSheetDialog
+            bottomSheet = sheetDialog?.findViewById<View>(
+                    com.google.android.material.R.id.design_bottom_sheet
+            ) as? FrameLayout
+        }
```

Comment:
Does it need to cater to landscape mode here?

Reply:
I know that during timzeone picker work I mentioned that the bottom sheet is collapsed in landscape mode, but after some research, it seems like its intended behavior 🤷